factory/base.py

```diff
@@ -157,6 +157,14 @@ def __init__(self):
         self.parameters = {}
         self.parameters_dependencies = {}
 
+    @property
+    def sorted_postgen_declarations(self):
+        """Get sorted postgen declaration items."""
+        return sorted(
+            self.postgen_declarations.items(),
+            key=lambda item: item[1].creation_counter,
```

Comment:
Sorted doesn't explode tuples (because it doesn't introspect the elements it is sorting), so unfortunately this is not possible. We have to deal with 1 object here.